.github/workflows/ci.yml

```diff
@@ -52,16 +48,12 @@ jobs:
   test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
-        with:
-          fetch-depth: 0
-      - uses: ./github-actions/setup-bazel-remote-exec
-        with:
-          bazelrc: .bazelrc
-      - uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # tag=v3.6.0
-        with:
-          node-version-file: '.nvmrc'
-          cache: 'yarn'
+      # Because the checkout and setup node action is contained in the dev-infra repo, we must
+      # checkout the repo to be able to run the action we have created.  Other repos will skip
+      # this step.
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # tag=v3.5.3
+      - uses: ./github-actions/utils/checkout-and-setup-node
+      - uses: ./github-actions/utils/setup-bazel-remote-exec
```

Comment:
One question: what's the context for moving these into `utils`? aren't all actions workflow utils?

Reply:
They are different as they are the building blocks for the environment and setup for github actions.

Typically you would only use one of the other actions we define here, but these ones are used in conjunction (as preparation) for all of the others.

Reply:
I see what you're seeing, but I consider other actions like for preview deployments similar buildings blocks- so not sure if the separate folder is a useful distinction- but that's really just my pers feeling

Reply:
I think I am just trying to find a categorization/structure that prevents us from having one directory with like 50 actions thats hard to search/understand.

Reply:
Why would the preview workflow build blocks not be part of this though? should we rather have more categorized folders? e.g.

* github-actions
    * bazel
    * npm
    * previews

Reply:
We could categorize like this if you would like.  I am not entirely 100% set on the best structure and happy to see where it can best end up as we go

Reply:
Updated to use your proposed directories.  We can follow up to categorize and structure/sort the others in follow ups.